docs/OpenTypeFeatureFileSpecification.md

```diff
@@ -1267,6 +1292,44 @@ rules, the implementation sorts the rules to avoid conflict; for example, the
 ligature substitution rule for f_f_i will be written before the ligature
 substitution rule for f_i, no matter what their order is in the feature file.
 
+<a name="4.e.i"></a>
+#### 4.e.i Feature Variations (variable font feature replacement)
+
+For variable fonts *only*, it is possible to specify [FeatureVariations](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#featurevariations-table), which allow the use of different lookups for different parts of the variation space.
+
+<a name="4.e.i.1"></a>
+##### 4.e.i.1 `conditionset`
+
+A named `conditionset` defines [conditions](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#conditionset-table) that can trigger a variable font lookup to swap in different lookups when the conditions are matched. A `conditionset` is simply a list of axis tags and min/max values. A conditionset is matched when all conditions are true. Syntax:
+```fea
+conditionset <name of set> {
+    <axis tag> <minValue> <maxValue>;
+    # additional tag min max entries
+} <name of set>;
+```
+For example, to define a "heavy" condition, for the 'wght' axis between 700-900:
+```fea
+conditionset heavy {
+    wght 700 900;
+} heavy;
+```
+
+<a name="4.e.i.2"></a>
+##### 4.e.i.2 `variation` (FeatureVariation)
+A `variation` is akin to a regular `feature` definition, with an added `conditionset` specifier to indicate that the feature should be active when the conditions of the conditionset are met. The syntax is:
```

Comment:
OT gives us feature table *substitutions* - in the sense that one feature table is swapped out and another is swapped in. So if you have:

```
feature test {
   sub a by b;
} test;

variation test heavy {
   sub d by e;
} test;
```

then if the heavy condition is met, the variation gets swapped in, the original feature gets swapped out, and *only* the `d->e` substitution will happen, as I understand it. 

If this is the intent, then this should be clearer in the text here. I suggest replacing "to indicate that the feature should active" (which is kind of wooly anyway - "active"?) with "to indicate that the feature should be replaced by the rules contained in the `variation` statement".

Reply:
> In fact, I think this is extremely cumbersome to use. Since the "varied feature" needs to list everything that applies at that point.

In other words, the current design would require would to call `fontTools.varLib.featureVars`-like code to resolve rules that apply at every region to generate the .fea file. Whereas in my proposal, the designer intentions are encoded directly and resolved when compiling the font.

cc @justvanrossum

Reply:
I missed this important point when I first reviewed this. The way we implemented int feaLib, by using featureVars.addFeartureVariationsRaw, means that the the new Feature table that replaces the default one when the conditions are met simply extends the existing feature's lookup list indices, but does not replaces it:

https://github.com/fonttools/fonttools/blob/7412268cbe2a36f588e2d06b45faf35a3b803633/Lib/fontTools/varLib/featureVars.py#L429-L436

Maybe we should try to support both use-cases somehow, making this explicit at the syntax level.